### ISSUE TITLE
Labels with additional non translation comments cause an error

### ISSUE THREAD

**Issue by hhfiddelke** [DanielNT/ALTranslationCenter#17]
if you have a label, f.e.:
`CodeunitErr: Label 'There is no processing Codeunit defined for Profile %1', Comment = 'DEU="Für Profil %1 ist keine Bearbeitungs-Codeunit definiert.", %1=Profile Code';`

with a comment that also explains a placeholder or something different, the translation tool fires an exception.

It should only use the the text that is formatted according to the translation syntax 'DEU="ABC",ENU="YXZ"' and ignore the rest.

**Comment by DanielNT** [comment]
Hi @hhfiddelke.

I tested with `Skip % only prefix` option, and your example does not work well, I think that valid syntax with "%" is something like:

`CodeunitErr: Label 'There is no processing Codeunit defined for Profile %1', Comment = 'DEU="Für Profil %1 ist keine Bearbeitungs-Codeunit definiert.",%1="Profile Code"';`

**Comment by hhfiddelke** [comment]
Thank you, for your tip, that seems to work.
Unfortunately does Microsoft not provide any documentation about how to format and structure comments, so the only documentation i found was [https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0470](url) does not recommend your description.

So i can live with the restriction, but if you can spent some time on improvements, it would be nice ATC would ignore anything in a comment beginning with the first % or not starting with a Language Code .

**Comment by DanielNT** [comment]
You are absolutely right, the recommended syntax for detailing "%" is not the same. The problem is that, for getting translations and detecting possible errors, the string must be splitted. The trouble could be that changing it may throw false positive errors, or split the translation incorrectly in some cases, if it includes `%1`, `%2`, etc. in translations.

I will check it when I have free time, and maybe add a new experimental option.

Thanks.

**Comment by DanielNT** [comment]
A new version (v2.7.0) is published, and now the recommended syntax can be used.

**Comment by JavierFuentes** [comment]
Hi @DanielNT ,
I've been doing some tests about this issue, and I've found with v2.7.1 of your extension that it works right only activating this other **checkForWhitespacesInLanguageAbbreviation:s: false** setting  because **throwErrorIfCommentLanguageNotMapped : "Skip only % prefix"** seems to do nothing right now:

```json5
{
	"alTranslationCenter.general.customLanguageMapping": {
			"ESP": "es-ES|es-ES_tradnl"
	},
	"alTranslationCenter.general.checkForWhitespacesInLanguageAbbreviations": false,
	// "alTranslationCenter.general.throwErrorIfCommentLanguageNotMapped": "Skip only % prefix",  // it seems not appliying and/or needed
}
```

Following the example of Hans you commented above too, with this settings this works right now for me:

```al
// Added a white space after the comma before %1, and drop double-quotes after %1=
CodeunitErr: Label 'There is no processing Codeunit defined for Profile %1', 
             Comment = 'ESP="Für Profil %1 ist keine Bearbeitungs-Codeunit definiert.", %1=profile-code';
```

**Comment by DanielNT** [comment]
Hi @JavierFuentes, with the property
```json
 "alTranslationCenter.general.throwErrorIfCommentLanguageNotMapped": "Skip only % prefix"
```
the comment
```al
// Added a white space after the comma before %1, and drop double-quotes after %1=
CodeunitErr: Label 'There is no processing Codeunit defined for Profile %1', 
             Comment = 'ESP="Für Profil %1 ist keine Bearbeitungs-Codeunit definiert.", %1=profile-code';
```

has a space before the %1, so it throws an error. If you set the property ```checkForWhitespacesInLanguageAbbreviations``` to false, it will work, because the default property in ```alTranslationCenter.general.throwErrorIfCommentLanguageNotMapped``` is **skipAll**, so it ignores the languages codes not mapped, included ```%1, %2, ...``` code (placeholders). If you set to **skipNone**, placeholders will throw a error too.

**Comment by JavierFuentes** [comment]
Hi @DanielNT ,
the point is, I tried configuring it as you indicated in your initial comment and adding the double quotes to the strings with placeholders, but although it didn't give any errors, they were later included in the translated string, so they didn't work correctly.

Of all the tests and combinations of settings I ran, the only one that worked correctly is the one I mentioned and `"Skip only % prefix"` didn't seem to do its job well because it either didn't prevent the error or simply didn't do anything that the other settings I used didn't already do.

**Comment by DanielNT** [comment]
@JavierFuentes, could you share a example project with the _settings.json_ which causes the translated string to be included not correctly? 
In my tests, the translations are generating as expected (placeholders ignored and translated strings without them).
